code/__HELPERS/dynamic_human_icon_gen.dm

```diff
@@ -0,0 +1,55 @@
+GLOBAL_LIST_EMPTY(dynamic_human_icons)
+
+/proc/get_dynamic_human_icon(outfit_path, species_path = /datum/species/human, mob_spawn_path, r_hand, l_hand, bloody_slots = NONE, list/generated_dirs = GLOB.cardinals)
+	if(!species_path)
+		return FALSE
+	var/arg_string = "[outfit_path]_[species_path]_[mob_spawn_path]_[l_hand]_[r_hand]_[bloody_slots]_[generated_dirs.Join(",")]"
+	if(GLOB.dynamic_human_icons[arg_string])
+		return GLOB.dynamic_human_icons[arg_string]
+	var/mob/living/carbon/human/dummy/consistent/dummy = new()
+	dummy.set_species(species_path)
+	dummy.stat = DEAD //this is to avoid side effects of mob spawners
+	dummy.underwear = "Nude"
+	dummy.undershirt = "Nude"
+	dummy.socks = "Nude"
+	if(outfit_path)
+		var/datum/outfit/outfit = new outfit_path()
+		if(r_hand != FALSE) //we can still override to be null, false means just use outfit's
+			outfit.r_hand = r_hand
+		if(l_hand != FALSE)
+			outfit.l_hand = l_hand
+		dummy.equipOutfit(outfit)
+	else if(mob_spawn_path)
+		var/obj/effect/mob_spawn/spawner = new mob_spawn_path(null, TRUE)
+		spawner.outfit_override = list()
+		if(r_hand != NO_REPLACE)
+			spawner.outfit_override["r_hand"] = r_hand
+		if(l_hand != NO_REPLACE)
+			spawner.outfit_override["l_hand"] = l_hand
+		spawner.special(dummy, dummy)
+		spawner.equip(dummy)
+	for(var/obj/item/carried_item in dummy)
+		if(dummy.is_holding(carried_item))
+			if(carried_item.GetComponent(/datum/component/two_handed))
+				dummy.swap_hand(dummy.get_held_index_of_item(carried_item))
+				carried_item.attack_self(dummy)
+			if(carried_item.GetComponent(/datum/component/transforming))
+				carried_item.attack_self(dummy)
+		if(bloody_slots & carried_item.slot_flags)
+			carried_item.add_mob_blood(dummy)
+	dummy.update_held_items()
+	var/icon/output = icon('icons/blanks/32x32.dmi', "nothing")
+	for(var/direction in generated_dirs)
+		var/icon/partial = getFlatIcon(dummy, defdir = direction)
+		output.Insert(partial, dir = direction)
+	GLOB.dynamic_human_icons[arg_string] = output
```

Comment:
Can you explain some of what you're doing with normal comments? And use some empty space because this is just a wall